internal/services/network/express_route_gateway_resource.go

```diff
@@ -138,6 +136,73 @@ func resourceExpressRouteGatewayCreateUpdate(d *pluginsdk.ResourceData, meta int
 	return resourceExpressRouteGatewayRead(d, meta)
 }
 
+func resourceExpressRouteGatewayUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Network.ExpressRouteGateways
+	connectionsClient := meta.(*clients.Client).Network.ExpressRouteConnections
+	subscriptionId := meta.(*clients.Client).Account.SubscriptionId
+	ctx, cancel := timeouts.ForUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	log.Println("[INFO] preparing arguments for ExpressRoute Gateway update.")
+
+	id := expressroutegateways.NewExpressRouteGatewayID(subscriptionId, d.Get("resource_group_name").(string), d.Get("name").(string))
```

Comment:
We can parse the ID here instead of generating a new one
```suggestion
	id, err := expressroutegateways.ParseExpressRouteGatewayID(d.Id())
	if err != nil {
		return err
	}
```

Reply:
thanks! fixed now 👍